pkg/storage/snapshot/restore.go

```diff
@@ -271,6 +271,10 @@ func (ctrl *VMRestoreController) reconcileVolumeRestores(vmRestore *snapshotv1.V
 		return false, err
 	}
 
+	snapshotVM := content.Spec.Source.VirtualMachine
```

Comment:
This betrays the restoreTarget interface. It should not be doing anything VM specific like dealing with datavolumetemplates.  I think it would be more appropriate to either:

1) Update volume restores + annotate PVCs in `restoreTarget.Reconcile()` sometime before generateRestoredVMSpec() is called.  I think it would be best if generateRestoredVMSpec does no work other than generate a VM spec

2) Add some new method to `restoreTarget` that is called in `reconcileVolumeRestores` that mutates the `VolumeRestore`. Still need to deal with annotating the PVCs though